src/localplayer.cpp

```diff
@@ -1077,3 +1074,81 @@ float LocalPlayer::getSlipFactor(Environment *env, const v3f &speedH)
 	}
 	return 1.0f;
 }
+
+void LocalPlayer::handleAutojump(f32 dtime, Environment *env,
+		const collisionMoveResult &result, const v3f &old_position,
+		const v3f &old_speed, f32 pos_max_d)
+{
+	PlayerSettings &player_settings = getPlayerSettings();
+	if (!player_settings.autojump)
+		return;
+
+	if (m_autojump) {
+		// release autojump after a given time
+		m_autojump_time -= dtime;
+		if (m_autojump_time <= 0.0f)
+			m_autojump = false;
+		return;
+	}
+
+	bool control_forward = control.up ||
+			       (!control.up && !control.down &&
+					       control.forw_move_joystick_axis < -0.05);
+	bool could_autojump =
+			m_can_jump && !control.jump && !control.sneak && control_forward;
+	if (!could_autojump)
+		return;
+
+	bool horizontal_collision = false;
+	for (const auto &colinfo : result.collisions) {
+		if (colinfo.type == COLLISION_NODE && colinfo.plane != 1) {
+			horizontal_collision = true;
+			break; // one is enough
+		}
+	}
+
+	// must be running against something to trigger autojumping
+	if (!horizontal_collision)
+		return;
+
+	// check for nodes above
+	v3f headpos_min = m_position + m_collisionbox.MinEdge * 0.99f;
+	v3f headpos_max = m_position + m_collisionbox.MaxEdge * 0.99f;
+	headpos_min.Y = headpos_max.Y; // top face of collision box
+	v3s16 ceilpos_min = floatToInt(headpos_min, BS) + v3s16(0, 1, 0);
+	v3s16 ceilpos_max = floatToInt(headpos_max, BS) + v3s16(0, 1, 0);
+	const NodeDefManager *ndef = env->getGameDef()->ndef();
+	bool is_position_valid;
+	for (s16 z = ceilpos_min.Z; z <= ceilpos_max.Z; z++) {
+		for (s16 x = ceilpos_min.X; x <= ceilpos_max.X; x++) {
+			MapNode n = env->getMap().getNodeNoEx(v3s16(x, ceilpos_max.Y, z), &is_position_valid);
+
+			if (is_position_valid && n.getContent() != CONTENT_IGNORE) {
```

Comment:
Ignore isn't 'walkable = true' though, but players collide with it.

Reply:
If they collide with it, then they should prevent autojump. I've refactored it so that invalid positions (outside the map?) do not prevent jumping, but ignore does, and got rid of some indentation in the process.